ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/layered/AbstractPrioritizedTransactions.java

```diff
@@ -137,6 +141,13 @@ protected void internalRemove(
     orderByFee.remove(removedTx);
   }
 
+  @Override
+  protected void internalPenalize(final PendingTransaction penalizedTx) {
+    orderByFee.remove(penalizedTx);
```

Comment:
TreeSet is not synchronized, so multiple subclasses could experience race issues. Suggest either making it sync or making this method sync.